### PR TITLE
TM-5672 update cycle date classifications

### PR DESCRIPTION
**Pull request by M00staff**
Dual Merge: 
- [FE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2950)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/463)

[Ticket](https://metaphase.atlassian.net/browse/TM-5672)

### CODE REVIEW

File: talentmap_api/fsbid/services/assignment_cycles.py [hunk 1]
```diff
@@ -550,3 +550,46 @@ def success_mapping(x):
...
+        'PV_API_VERSION_I': '',
+        'PV_AD_ID_I': '',
+        'I_CYCLE_ID': cycle_id,
+        'I_CDT_CD': cycle_code,
+        'I_INC_IND': checked_values_string,
+        'I_PCT_CODE': checked_codes_string,
+        'I_CDC_UPDATE_ID': ',,,,,,,,,,,,,,',
+        'I_CDC_UPDATE_DATE': ',,,,,,,,,,,,,,',
+    }
+    return mapped_request
```
Comment (M00staff) [inline talentmap_api/fsbid/services/assignment_cycles.py:595]: I jimmied this all to make it simpler (can only update one cycle at a time (avoids having to do anything bonkers with regard to figuring out which `I_CYCLE_IND` & `I_CDT_CD` were updated 

also, it just passes in all of the value for that cycle - vs just the ones that were updated  🤷

... i DEFINITELY need to test this in dev 🤔 
(and swagger - it was late sorry!)

Spreadsheet Example:
```
{
"PV_API_VERSION_I": "",
"PV_AD_ID_I": "",
"I_INC_IND": "1,1,1,1,1",
"I_CYCLE_ID": "228,228,227,229,229",
"I_CDT_CD": "BUREARLY,BUREARLY,BURPREBD,BURPREBD,BUREARLY",
"I_PCT_CODE": "15,CN,15,DC,DC",
"I_CDC_UPDATE_ID": ", , , ,",
"I_CDC_UPDATE_DATE": ", , , ,",
"O_RETURN_CODE": "",
"QRY_ACTION_DATA": "",
"QRY_ERROR_DATA": ""
}
```

Print Response 1
```
{
'PV_API_VERSION_I': '', 
'PV_AD_ID_I': '', 
'I_CYCLE_ID': 217, 
'I_CDT_CD': 'BUREARLY', 
'I_INC_IND': '0,1,0,0,1,0,0,0,0,0,0,0,0,1', 
'I_PCT_CODE': '15,CN,DC,FI,HD,IT,KY,MC,PD,SC,SN,SO,XR,XX', 
'I_CDC_UPDATE_ID': ',,,,,,,,,,,,,,', 
'I_CDC_UPDATE_DATE': ',,,,,,,,,,,,,,'
}
```

2nd Example 
```
{
'PV_API_VERSION_I': '',
'PV_AD_ID_I': '',
'I_CYCLE_ID': 225,
'I_CDT_CD': 'BUREARLY',
'I_INC_IND': '0,1,0,0,0,0,0,1,0,0,0,0,1,1',
'I_PCT_CODE': '15,CN,DC,FI,HD,IT,KY,MC,PD,SC,SN,SO,XR,XX',
'I_CDC_UPDATE_ID': ',,,,,,,,,,,,,,',
'I_CDC_UPDATE_DATE': ',,,,,,,,,,,,,,'
}
```

Comment (M00staff) [reply talentmap_api/fsbid/services/assignment_cycles.py]: bonked a few things above 
- have to repeat `I_CDT_CD` and `I_CYCLE_ID` (and make sure its a string) 
- make sure to pass through `I_CDC_UPDATE_ID` and `I_CDC_UPDATE_DATE` - and if the value is 0 or null, make sure to pass in a blank string that's comma separated

```
{
  'PV_API_VERSION_I': '',
  'PV_AD_ID_I': '',
  'I_CYCLE_ID': '227,227,227,227,227,227,227,227,227,227,227,227,227,227',
  'I_CDT_CD': 'BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD,BURPREBD',
  'I_INC_IND': '1,0,0,0,0,0,0,0,0,0,0,0,1,0',
  'I_PCT_CODE': '15,CN,DC,FI,HD,IT,KY,MC,PD,SC,SN,SO,XR,XX',
  'I_CDC_UPDATE_ID': '57675,,,,,,,,,,,,,',
  'I_CDC_UPDATE_DATE': '20240402083632,,,,,,,,,,,,,'
}
```

actually pretty nifty & fun to do 

working now - checked in dev

File: talentmap_api/fsbid/services/assignment_cycles.py [hunk 2]
```diff
@@ -550,3 +550,56 @@ def success_mapping(x):
...
+    data = req['data']
+    cycle_ids = [str(data['id']) for _ in data['values']]
+    cycle_codes = [data['code'] for _ in data['values']]
+    checked_values = [item['value'] for item in data['values']]
+    checked_codes = [item['code'] for item in data['values']]
+    # if the date or id value is null or zero pass in blank string so:
+    # 'I_CDC_UPDATE_ID': '57675,,,,,,,,,,,,,',
+    # 'I_CDC_UPDATE_DATE': '20240402083632,,,,,,,,,,,,,'
+    checked_dates = [(str(item['update_date']) if item['update_date'] else "") for item in data['values']]
+    checked_ids = [(str(item['update_id']) if item['update_id'] else "") for item in data['values']]
```
Comment (jlam-mpc) [inline talentmap_api/fsbid/services/assignment_cycles.py:586]: Nice way of handling this on the backend. I may end up adapting Position Classifications and all of the other similar update requests to do the same in the future. Unless we later decide we want to standardize it on the frontend instead 🤔. Going to leave it alone for now-

File: talentmap_api/fsbid/services/assignment_cycles.py [hunk 3]
```diff
@@ -550,3 +550,56 @@ def success_mapping(x):
         }
 
     return service_response(data, 'Cycle Classifications Data', success_mapping)
+
+
+def update_assignment_cycle_classification_dates(jwt_token, request):
```
Comment (jlam-mpc) [inline talentmap_api/fsbid/services/assignment_cycles.py:555]: Not sure if it matters to you but just wanted to note the different naming convention here ^
update_assignment_cycle_classification_dates
update_assignment_cycles_classifications_req_mapping
update_assignment_cycles_classifications_res_mapping

**Review by jlam-mpc — APPROVED** [review]

**Pull request merged**